tests/test-bootstrap.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
+set -euo pipefail
+
+docker build -t eks-optimized-ami -f "${SCRIPTPATH}/Dockerfile" "${SCRIPTPATH}/../"
+
+summary=""
+status=0
+
+function run(){
+    local exit_code=0
+    ## expectation status code
+    local expectation=$1
+    shift
+    ## test name for output
+    local name=$1
+    shift
+    echo "#########################################################################################################"
+    echo "$@"
+    echo "---------------------------------------------STDOUT------------------------------------------------------"
+    docker run -v ${PWD}/../files/bootstrap.sh:/etc/eks/bootstrap.sh \
+        -v ${PWD}/../files/max-pods-calculator.sh:/etc/eks/max-pods-calculator.sh \
+        -it eks-optimized-ami $@ || exit_code=$?
+    if [[ ${exit_code} -eq ${expectation} ]]; then
+        msg="✅ Test \"${name}\" Passed"
+    else
+        msg="❌ Test \"${name}\" Failed"
+        status=1
+    fi
+    summary+="${msg}\n"
+    echo "${msg}"
+    echo -e "#########################################################################################################\n\n"
+}
+
+run 0 "Default params should pass" /etc/eks/bootstrap.sh \
+    --b64-cluster-ca dGVzdA== \
+    --apiserver-endpoint http://my-api-endpoint \
+    test
+
+run 1 "Should fail w/ \"service-ipv6-cidr must be provided when ip-family is specified as ipv6\"" /etc/eks/bootstrap.sh \
+    --b64-cluster-ca dGVzdA== \
+    --apiserver-endpoint http://my-api-endpoint \
+    --ip-family ipv6 \
+    test
+  
+run 0 "Should return IPv6 DNS cluster IP when given service-ipv6-cidr" /etc/eks/bootstrap.sh \
```

Comment:
So far as I can tell, we only validate exit_code here. We are not actually testing that IPv6 value is set for DNS cluster IP. If this test was executed against code in https://github.com/awslabs/amazon-eks-ami/pull/860 (where we had bug setting DNS cluster ip for ipv6), I believe it would pass. Can we update these tests to actually validate DNS cluster IP value? 

Even if that is just manually looking at a log line, it would be helpful.

Reply:
Yeah, that's a good idea. It was mainly a limitation of time figuring out how to pipe inspect the files after a run. But I think I have a solution that works okay, at least for checking kubelet-config.json after a bootstrap run.